eth/contracts/Most.sol

```diff
@@ -191,37 +230,42 @@ contract Most is
             return;
         }
 
-        require(_requestHash == requestHash, "Hash does not match the data");
+        if (_requestHash != requestHash) revert DataHashMismatch();
 
         request.signatures[msg.sender] = true;
-        request.signatureCount++;
+        ++request.signatureCount;
 
         emit RequestSigned(requestHash, msg.sender);
 
-        if (request.signatureCount >= signatureThreshold[committeeId]) {
+        if (request.signatureCount >= signatureThreshold[_committeeId]) {
             processedRequests[requestHash] = true;
             delete pendingRequests[requestHash];
 
+            address _destTokenAddress = bytes32ToAddress(destTokenAddress);
+            address _destReceiverAddress = bytes32ToAddress(
+                destReceiverAddress
+            );
             // return the locked tokens
-            if (bytes32ToAddress(destTokenAddress) == wethAddress) {
+            if (_destTokenAddress == wethAddress) {
                 (bool unwrapSuccess, ) = wethAddress.call(
-                    abi.encodeWithSignature("withdraw(uint256)", amount)
-                );
-                require(
-                    unwrapSuccess,
-                    "Failed to uwrap wrapped ETH to native ETH."
-                );
-                (bool sendNativeEthSuccess, ) = bytes32ToAddress(
-                    destReceiverAddress
-                ).call{value: amount}("");
-                require(
-                    sendNativeEthSuccess,
-                    "Failed to send the native ETH back to the user."
+                    abi.encodeCall(IWETH9.withdraw, (amount))
                 );
+                if (!unwrapSuccess) revert UnwrappingEth();
+                (bool sendNativeEthSuccess, ) = _destReceiverAddress.call{
+                    value: amount,
+                    gas: GAS_LIMIT
+                }("");
+                if (!sendNativeEthSuccess) {
+                    if (isContract(_destReceiverAddress)) {
```

Comment:
If the receiver is a contract and the transfer fails, there's a high chance that 
- it doesn't implement `receive()`/`fallback(...)`
- `receive()`/`fallback(...)` consumes more than 3500 gas

In both cases, we don't want to revert because relayers would retry this operation indefinitely.
If the receiver is EOA, we want to retry the transfer (relayer calls `receiveRequest(...)` again).

Reply:
Ok, makes sense. But documentation should have BIG WARNING that SC has to use <= 3500 gas on fallback from bridge